File: tests/fixtures/worker-pool/node/slow/handler.js
function sleep(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

exports.handler = async (_event) => {
  await sleep(900);
  return {
    status: 200,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ ok: true, runtime: "node" }),
  };
};

File: tests/fixtures/worker-pool/rust/slow-rust/handler.rs
use serde_json::{json, Value};
use std::{thread, time::Duration};

pub fn handler(_event: Value) -> Value {
    thread::sleep(Duration::from_millis(200));
    json!({
        "status": 200,
        "headers": { "Content-Type": "application/json" },
        "body": json!({ "ok": true, "runtime": "rust" }).to_string()
    })
}

